sql/messaging_system_derived/cfr_active_users_v1/query.sql

```diff
@@ -0,0 +1,96 @@
+-- This calculates the daily, weekly, monthly active clients based on the pings
+-- in `messaging_system_stable.cfr_v1`. A 7-day averaged daily is also calculated.
+-- The results are inserted into `messaging_system_derived.cfr_active_clients_v1`.
+-- Phase 1: DELETE the data for @submission_date if any.
+DELETE FROM
+  cfr_active_clients_v1
+WHERE
+  submission_date = @submission_date;
+
+-- Phase 2: INSERT daily, weekly (7 days), and monthly (28 days) for @submission_date
+WITH _daily AS (
+  SELECT
+    CAST(@submission_date AS DATE) AS submission_date,
+    release_channel,
+    -- The `release` channel sends `impression_id`, whereas the prerelease channels send client_id.
+    COUNT(DISTINCT IFNULL(impression_id, client_id)) AS daily,
+  FROM
+    messaging_system_stable.cfr_v1
+  WHERE
+    EXTRACT(date FROM submission_timestamp) = @submission_date
+  GROUP BY
+    1,
+    2
+),
+_weekly AS (
+  SELECT
+    CAST(@submission_date AS DATE) AS submission_date,
+    release_channel,
+    COUNT(DISTINCT IFNULL(impression_id, client_id)) AS daily,
+  FROM
+    messaging_system_stable.cfr_v1
+  WHERE
+    EXTRACT(date FROM submission_timestamp)
+    BETWEEN DATE_SUB(@submission_date, INTERVAL 6 DAY)
+    AND @submission_date
+  GROUP BY
+    1,
+    2
+),
+_monthly AS (
+  SELECT
+    CAST(submission_date AS DATE) AS submission_date,
+    release_channel,
+    COUNT(DISTINCT IFNULL(impression_id, client_id)) AS daily,
+  FROM
+    messaging_system_stable.cfr_v1
+  WHERE
+    EXTRACT(date FROM submission_timestamp)
+    BETWEEN DATE_SUB(@submission_date, INTERVAL 27 DAY)
+    AND @submission_date
+  GROUP BY
+    1,
+    2
+)
+INSERT INTO
```

Comment:
Wherever possible, we like to structure queries in this repository as simple `SELECT` statements that represent a single day partition of the target table. The machinery in Airflow will run the query with `destination_table` set to be the individual partition of the table corresponding to the target day. That handles reprocessing issues (existing content for that day will be overwritten).

So we generally expect not to see DELETE or INSERT in our incremental queries in this repository.